x/attribute/keeper/keeper.go

```diff
@@ -439,10 +450,15 @@ func (k Keeper) PurgeAttribute(ctx sdk.Context, name string, owner sdk.AccAddres
 		return err
 	}
 	for _, acct := range accts {
+		attrToDelete := [][]byte{}
 		store := ctx.KVStore(k.storeKey)
 		it := sdk.KVStorePrefixIterator(store, types.AddrAttributesNameKeyPrefix(acct, name))
 		for ; it.Valid(); it.Next() {
-			store.Delete(it.Key())
+			attrToDelete = append(attrToDelete, it.Key())
+		}
+		it.Close()
```

Comment:
This needs to be extracted to its own function so that you can ensure the iterator is always properly closed after each use. The reason you always want a `defer` that closes the iterator, is so that it gets closed even if there's a panic.

But because this is in a for loop, those defers wouldn't be called until much later. It'd probably also be safe to do the `if it != nil { it.Close() }` way on it and leave it in the loop, but it's probably cleaner to just extract it to its own function. I'd have the function take in a `KVStore` instead of the `ctx` too, so you only need to create that store once.